bf/bf.c

```diff
@@ -29,48 +27,78 @@ int main(int argc, char const *argv[])
 	else
 	{
 		FILE *bf_file = fopen(argv[1], "r");
-		FILE *c_output = fopen(argv[2], "w");
+        	int matches = 0;
 		char c;
-
-		fputs("/* generated by bf.c */#include <stdio.h>\nint main(void) {char s[30000]={0};char*sp=s;", c_output);
+		size_t size = 1;
+		char *op_buffer = malloc(EIGHT_KB);
+		assert(op_buffer != NULL);
 
 		while ((c = fgetc(bf_file)) != EOF)
 		{
+			if (size % EIGHT_KB)
```

Comment:
This will cause a reallocation whenever `size` is not a mulitple of `EIGHT_KB`, which I don't think is what you intended.

What we really want is a reallocation whenever size is getting close to the allocated buffer size. Maybe something like this?

```C
// 14 is our largest sequence
if (size + 14 >= buf_max) {
  buf_max *= 2;
  op_buffer = realloc(op_buffer, buf_max);
  // check op_buffer to make sure allocation went alright
}
```

(That can probably be improved).